### DISCUSSION TITLE
How to show current shell in use?

### DISCUSSION THREAD

**Discussion by nerd190** [starship/starship#1323] [General]
First, **THANK YOU** for this simple, effective (and beautiful!) prompt.

I have read through the site on configuration but cannot find how to show the current shell in use...
(or I missed it!)

**Why?**

I would like to use this prompt for the three shells on my system: bash, zsh & fish **but** I want them to state what the current shell is, so I actually know what I am using, if all three share the same prompt, I will likely forget what shell I am in! 😆 the prompts each shell use have always shown this clearly as they each differ.

**Comment by matchai** [comment]
Starship sets the `STARSHIP_SHELL` to the currently running shell.

For something simple, you could use the [`env_var`](https://starship.rs/config/#environment-variable) module to show the current value:
```rust
[env_var]
variable = "STARSHIP_SHELL"
```

If you want something a little more custom (e.g. show a specific emoji for the current shell), that can be achieved with a [`custom`](https://starship.rs/config/#custom-commands) module.

**Reply by nerd190** [reply]
**THANK YOU!!!** @matchai 
Following this question, using `starship config` takes me to vim and shows as a "new file" so obviously, no contents (as it does not exist yet), so I assume I am using starship's default settings as nothing is declared... If I add the above `env_var` snippet alone:
- Would this mess up the default settings and just show my shell version in the prompt?
- Or would it **add** this to the default prompt?

**Reply by matchai** [reply]
Any configuration overrides the defaults for the configured value.
In the case of `env_var` and `custom`, there are no defaults so the changes would only be additive.

**Reply by andytom** [reply]
If you are looking for an example using the custom module one dalefrancum added an example in [Share your custom commands/modules!]( https://github.com/starship/starship/discussions/1252#discussioncomment-18316).

**Reply by nerd190** [reply]
Damn! This is nice!!! The original solution using `env_var`gave me:

**~/scripts** on **bash**

Not as beautiful :'(
So I was going to ask how to change it to:

**bash: ~/scripts**

Which looks prettier for sure!

**But** after seeing this! I have to... *its stunning!*

Thanks to **both** of you for helping, its truly appreciated.
Starship is a **very** clever prompt solution with great logic! (i.e using just `env_var` alone in doesnt mess with default, just adds it as it hasn't been declared before) with (it appears) very active, helpful members!
**Thank you!** @matchai & @andytom for solving this (personal) problem of forgetting which shell I am in! 🤣 
Finally... If one wanted the custom emoji *before* my path, instead of  *after* it, can this be achieved also?
🙄 *just being fussy now!* 🙄

**Reply by matchai** [reply]
To rearrange the modules in your prompt, you can use the [`prompt_order`](https://starship.rs/config/#prompt) configuration option. I would suggest copying the [default prompt order](https://starship.rs/config/#default-prompt-order) and moving `custom` to the start.
